src/main/java/edu/illinois/library/cantaloupe/cache/FilesystemCache.java

```diff
@@ -508,8 +508,8 @@ Set<Path> getDerivativeImageFiles(Identifier identifier)
                 hashedPathFragment(identifier.toString()));
         final String expectedNamePrefix =
                 StringUtils.md5(identifier.toString());
-        try {
-            return Files.list(cachePath)
+        try (final var fileStream = Files.list(cachePath)) {
+            return fileStream
                     .filter(p -> p.getFileName().toString().startsWith(expectedNamePrefix))
```

Comment:
@bbpennel. Nice! so the try() will auto close the fileStream... But... is the filter needed? Given the way caches are structured all files inside cachePath will need to be returned. Maybe for sanity just check if the file is regular file?
.filter(`Files::isRegularFile`) or something like that? I mean, just to save a few milliseconds per cycle... specially on large images with tons of derivatives

Reply:
The filter isn't a change in this PR, i didn't really evaluate whether it was needed or not. I'm fine with removing it if there isn't any concern of the cache path containing files that don't begin with the md5.

Reply:
@bbpennel thanks, true!, for the sake of not adding any point of rupture and because I don't know what use case the original Developer might had had in place, let's keep it! Approving. Thanks!